java/dataset/src/main/cpp/jni_wrapper.cc

```diff
@@ -261,6 +264,26 @@ void JNI_OnUnload(JavaVM* vm, void* reserved) {
   default_memory_pool_id = -1L;
 }
 
+/// Iterate over an object array of Tables Name on position `i` and theirs
+/// Memory Address representation on `i+1` position linearly.
+/// Return a mapping of the Table Name to Query as a Key and a RecordBatchReader
+/// as a Value.
+std::unordered_map<std::string, std::shared_ptr<arrow::Table>> ToMapTableToArrowReader(JNIEnv* env, jobjectArray& str_array) {
+  std::unordered_map<std::string, std::shared_ptr<arrow::Table>> map_table_to_record_batch_reader;
+  int length = env->GetArrayLength(str_array);
+  std::shared_ptr<arrow::Table> output_table;
+  for (int pos = 0; pos < length; pos++) {
```

Comment:
This needs to check if the array length is odd

Reply:
The JNI method is still accessible and I don't see why we can't have the proper checks here.

Reply:
ok, let me add that